src/modules/dashboard/sagas/actions/mintTokensMotion.tsx

```diff
@@ -0,0 +1,154 @@
+import {
+  createTransaction,
+  createTransactionChannels,
+  getTxChannel,
+} from '../../../core/sagas';
+import { ipfsUpload } from '../../../core/sagas/ipfs';
+import {
+  transactionReady,
+  transactionPending,
+  transactionAddParams,
+} from '../../../core/actionCreators';
+
+function* createMintTokensMotion({
+  payload: { colonyAddress, colonyName, amount, annotationMessage },
+  meta: { id: metaId, history },
+  meta,
+}: Action<ActionTypes.COLONY_MOTION_MINT_TOKENS>) {
+  let txChannel;
+  try {
+    if (!amount) {
+      throw new Error(
+        'Amount to mint not set for mintTokensMotion transaction',
+      );
+    }
+
+    const context = TEMP_getContext(ContextModule.ColonyManager);
+    const colonyClient = yield context.getClient(
+      ClientType.ColonyClient,
+      colonyAddress,
+    );
+
+    const encodedAction = colonyClient.interface.functions.mintTokens.encode([
+      amount,
+    ]);
+
+    /* Root domain is alaways 1 */
+    const { skillId } = yield call([colonyClient, colonyClient.getDomain], 1);
+
+    const { key, value, branchMask, siblings } = yield call(
+      colonyClient.getReputation,
+      skillId,
+      AddressZero,
+    );
+
+    txChannel = yield call(getTxChannel, metaId);
+
+    // setup batch ids and channels
+    const batchKey = 'createRootMotion';
+
+    const {
+      createRootMotion,
+      annotateMintTokens,
+    } = yield createTransactionChannels(metaId, [
+      'createRootMotion',
+      'annotateMintTokens',
+    ]);
+
+    // create transactions
+    yield fork(createTransaction, createRootMotion.id, {
+      context: ClientType.VotingReputationClient,
+      methodName: 'createRootMotion',
+      identifier: colonyAddress,
+      params: [AddressZero, encodedAction, key, value, branchMask, siblings],
+      group: {
+        key: batchKey,
+        id: metaId,
+        index: 0,
+      },
+      ready: false,
+    });
+
+    if (annotationMessage) {
+      yield fork(createTransaction, annotateMintTokens.id, {
+        context: ClientType.ColonyClient,
+        methodName: 'annotateTransaction',
+        identifier: colonyAddress,
+        params: [],
+        group: {
+          key: batchKey,
+          id: metaId,
+          index: 1,
+        },
+        ready: false,
+      });
+    }
+
+    yield takeFrom(createRootMotion.channel, ActionTypes.TRANSACTION_CREATED);
+    if (annotationMessage) {
+      yield takeFrom(
+        annotateMintTokens.channel,
+        ActionTypes.TRANSACTION_CREATED,
+      );
+    }
+
+    yield put(transactionReady(createRootMotion.id));
+
+    const {
+      payload: { hash: txHash },
+    } = yield takeFrom(
+      createRootMotion.channel,
+      ActionTypes.TRANSACTION_HASH_RECEIVED,
+    );
+    yield takeFrom(createRootMotion.channel, ActionTypes.TRANSACTION_SUCCEEDED);
+
+    if (annotationMessage) {
+      yield put(transactionPending(annotateMintTokens.id));
+
+      let ipfsHash = null;
```

Comment:
why let in here?

Reply:
It's used like this in all other sagas, so I just implemented the same. But it's a good question indeed... myabe @rdig  can tell us if it's needed. And I can remove if not.

Reply:
I mean, the explanation for this is right on the next line :)

https://github.com/JoinColony/colonyDapp/blob/eed7ed60e0b52c7f5c693b117b6af5eaf0aafd6c/src/modules/dashboard/sagas/actions/mintTokensMotion.tsx#L118-L123

The thing is that of the IPFS upload call fails it never sets a value to `ipfsHash` and when you try to call the contract method with the `ipfsHash` param being `undefined`, it will revert with an error.

But if it's `null` the call will succeed (obviously without an ipfs hash being set)

This is just basic fallback method to ensure that the call will always work, even if IPFS fails.

We'll improve on this in the future, but for the time being, this is good enough,

Reply:
I suspected something like this but was not sure. Thank you for the explanation!